samanthas_telegram_bot/data_structures/models_helper_classes.py

```diff
@@ -0,0 +1,90 @@
+"""Various dataclasses, NamedTuples and TypedDicts for attribute hinting and easy unpacking.
+
+Most of the classes correspond to models in the backend.
+
+The module is called helper_classes to distinguish it from context_types module that also contains
+custom classes for attribute hinting, but related to inner workings of python-telegram-bot.
+"""
+from dataclasses import dataclass
+from typing import NamedTuple, Optional, TypedDict
+
+from samanthas_telegram_bot.data_structures.enums import AgeRangeType, SmalltalkTestStatus
+
+
+@dataclass
+class AgeRange:
+    id: int
+    age_from: int
+    age_to: int
+    type: AgeRangeType
+    bot_phrase_id: Optional[str] = None
+
+
+@dataclass
+class AssessmentAnswer:
+    question_id: int
+    answer_id: int
+
+
+class AssessmentQuestionOption(NamedTuple):
+    id: int
+    text: str
+
+    def means_user_does_not_know_the_answer(self) -> bool:
+        return "i don't know" in self.text.lower()
+
+
+class AssessmentQuestion(NamedTuple):
+    id: int
+    text: str
+    options: tuple[AssessmentQuestionOption, ...]
+
+
+class Assessment(NamedTuple):
+    id: int
+    age_range_ids: tuple[int, ...]
+    questions: tuple[AssessmentQuestion, ...]
+
+
+@dataclass
+class DayAndTimeSlot:
+    id: int
+    day_of_week_index: int
+    from_utc_hour: int
+    to_utc_hour: int
+
+
+class LanguageAndLevel(NamedTuple):
+    id: int
+    language_id: str
+    level: str
+
+
+# using TypedDict for phrases to allow simple dict-like usage with dynamically determined locale
+class MultilingualBotPhrase(TypedDict):
+    en: str
+    ru: str
+    ua: str
+
+
+class SmalltalkResult(NamedTuple):
```

Comment:
Some of classes are TypedDict, some are NamedTuple and some are dataclass. Why so?

Reply:
Here's the logic:
1. The first choice is `NamedTuple`
2. If data structure must be mutable, `dataclass`.
3. If I need to dynamically get values by the key name, I choose `TypedDict` to avoid `getattr(phrase, locale)` syntax.

Reply:
How about `@dataclass(frozen=True)`? just to be consistent.

Reply:
I mean for the first case

Reply:
Done